### DISCUSSION TITLE
Why no 'single-quoted strings'

### DISCUSSION THREAD

**Discussion by eugenesvk** [kdl-org/kdl#327] [General]
I'm curious why "regular" strings are only "double-quoted" and not alternatively 'single-quoted'?

The single quotes have a tiny edge - they're less visible :) and also don't require <kbd>⇧</kbd> by default, so would be interesting to know what the design considerations were that excluded them

**Comment by tabatkins** [comment]
Double quotes *tend* to be considered the canonical quoting character when there's a choice, in many languages. And the main reason to have both is to let you avoid needing to use an escape when you want to write something with a double-quote in it, but that's avoided by the use of a raw string in KDL.

Notably, JSON also requires the use of double-quotes.

**Reply by eugenesvk** [reply]
> And the main reason to have both is to let you avoid needing to use an escape when you want to write something with a double-quote in it, but that's avoided by the use of a raw string in KDL.

Raw strings are great (and I think using a literal prefix `r` to signal type and a different symbol `#` instead of quote repetition a better design), but then there are the two other benefits I mentioned on the **input** and **output** side that single quotes have, so unless it breaks something would be nice to also have them


> Notably, JSON also requires the use of double-quotes.

and it's an even bigger issue with JSON since it has more quotes (for keys). At leas in KDL you have non-quoted nodes

**Comment by Diegovsky** [comment]
I would love to have this. It honestly saves me a lot of typing tbh.

**Reply by zkat** [reply]
It’s worth noting that kdl v2 already supports quoteless strings. That will probably save you more typing in the long run than not having to hold shift :)

**Reply by Diegovsky** [reply]
hey good to know! I'm a bit OOTL here: when is it releasing?

**Reply by zkat** [reply]
There's already two implementations with prerelease implementations of kdl v2 (`kdl-rs` for Rust, and `ckdl`  for Python, C, C++). See https://github.com/kdl-org/kdl/issues/372

Once we've worked out a couple of final items, we'll decide whether it's time to officially tag v2. There's only a couple of open questions we want to make sure we've got resolved, now that we have people who have actually implemented it and tried it out.
